source/installer.py

```diff
@@ -853,12 +864,12 @@ def _deleteFileGroupOrFail(
 			log.debugWarning(f"Failed to remove temp directory {tempDir}", exc_info=True)
 
 
-def install(shouldCreateDesktopShortcut: bool = True, shouldRunAtLogon: bool = True):
+def install(shouldCreateDesktopShortcut: bool = True, shouldRunAtLogon: bool = True) -> None:
 	prevInstallPath = WritePaths.installDir
 	installDir = WritePaths.defaultInstallDir
-	installDirX86 = WritePaths._installDirX86
+	installDirX86 = WritePaths._installDirX86 or WritePaths._defaultInstallDirX86
 	startMenuFolder = WritePaths.defaultStartMenuFolder
-	shouldCleanX86 = os.path.isdir(installDirX86)
```

Comment:
Perhaps for now this should temporarily  also check that NVDA itself is not x86? Otherwise once this is merged, and we are still x86, then the installer is going to try and remove / unregister each of the same things twice.  Or is the plan to switch to amd64 at the same time this is merged?

Reply:
@michaelDCurran - this is from an outdated diff, I think we have better handling for this now but let me know what you think